File: android/app/src/main/java/com/airbridge/app/core/models/DeviceInfo.kt
package com.airbridge.app.core.models

/** Device type classification — mirrors Windows enum. */
enum class DeviceType {
    WINDOWS_PC,
    ANDROID_PHONE,
    ANDROID_TABLET
}

/**
 * Represents a discovered or paired remote device.
 * Immutable data class — update by copy.
 */
data class DeviceInfo(
    val deviceId: String,
    val deviceName: String,
    val deviceType: DeviceType,
    val ipAddress: String,
    val port: Int,
    val isPaired: Boolean
)

File: windows/AirBridge.Core/Models/DeviceInfo.cs
namespace AirBridge.Core.Models;

/// <summary>Device type classification.</summary>
public enum DeviceType
{
    Unknown,
    WindowsPc,
    AndroidPhone,
    AndroidTablet
}

/// <summary>Represents a discovered or paired remote device.</summary>
public sealed record DeviceInfo(
    string DeviceId,
    string DeviceName,
    DeviceType DeviceType,
    string IpAddress,
    int Port,
    bool IsPaired
);
